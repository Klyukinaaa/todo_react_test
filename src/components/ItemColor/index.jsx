import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

function ItemColor(props) {
  const {
    i, item, clickColor, click,
  } = props;
  return (
    <span role="presentation" className={`btn ${click ? 'active' : ' '}`} onClick={() => clickColor(i)}>
      <input id="button" type="button" style={{ backgroundColor: item.backgroundColor }} />
    </span>
  );
}

ItemColor.propTypes = {
  i: PropTypes.number.isRequired,
  item: PropTypes.oneOfType([PropTypes.object]).isRequired,
  clickColor: PropTypes.func.isRequired,
  click: PropTypes.bool.isRequired,
};

export default ItemColor;
