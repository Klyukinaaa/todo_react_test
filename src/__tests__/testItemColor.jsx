import React from 'react';
import renderer from 'react-test-renderer';
import ItemColor from '../components/ItemColor';

test('renders correctly', () => {
  const color = {
    backgroundColor: '#ef666c',
    selected: true,
  }
  const tree = renderer
    .create(<ItemColor i={1} item={color} clickColor={() => void 0} click={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
