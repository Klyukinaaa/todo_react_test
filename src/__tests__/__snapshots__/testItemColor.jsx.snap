// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="btn  "
  onClick={[Function]}
  role="presentation"
>
  <input
    id="button"
    style={
      Object {
        "backgroundColor": "#ef666c",
      }
    }
    type="button"
  />
</span>
`;
